src/averageAges.js

```diff
@@ -37,7 +46,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(she => she.sex === 'f');
+  const moms = women.filter(she => people.some((child) => {
+    return child.mother === she.name;
+  }));
+
+  return withChildren
+    ? calculateAverageAge(moms)
+    : calculateAverageAge(women);
```

Comment:
first get the array based on the withChildren argument and then use `calculateAverageAge` on this array